### PR TITLE
Test Origin Login APIs

### PR DESCRIPTION
**Pull request by joereuss12**
This is a unit test for the origin login api's. The tests have been ran on both windows and in a linux container and shows everything running successfully. This effectively tests:
1. Code-based login when no password is set
2. Password reset API
3. Password-based login API
4. "whoami" API

### CODE REVIEW

**Comment by bbockelm** [comment]
Changing the reviewer over to Emma.

File: origin_ui/origin_test.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+		recorder := httptest.NewRecorder()
+		router.ServeHTTP(recorder, req)
+		//Check http reponse code 401
+		assert.Equal(t, 401, recorder.Code)
+		assert.JSONEq(t, `{"error":"Login failed"}`, recorder.Body.String())
+	})
+
+	//Invoke with invalid user, should fail
+	t.Run("Without user", func(t *testing.T) {
```
Comment (turetske) [inline origin_ui/origin_test.go:274]: You have tests for "Without user" and "Without password" but what about "Incorrect user" (where the user doesn't exist) or "Wrong password"

**Review by turetske — CHANGES_REQUESTED** [review]
LG except for possibly some changes in how temp files are being created and the addition of two more test cases as discussed in the specific comments.

File: origin_ui/origin_test.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+var (
+	tempPasswdFile *os.File
+	router         *gin.Engine
+)
+
+func TestMain(m *testing.M) {
+	gin.SetMode(gin.TestMode)
+
+	//set a temporary password file:
+	tempPasswdFile, err := os.CreateTemp("", "origin-ui-passwd")
```
[the remark this answers is not in the document]
Comment (joereuss12) [reply origin_ui/origin_test.go:46]: From investigating, it looks like I can't use the packages TempDir() function (I did not know this existed but that's helpful to know now) in the TestMain() function and to me it doesn't make too much sense to create a test file for each function if it is just needed for general setup and use throughout the other functions. I did however, change `defer tempPasswdFile.Close()` to `defer os.Remove(tempPasswdFile.Name())` and that looks like the file is being cleaned up and removed after all of the tests have run. Working to add the other tests you mentioned now!

File: origin_ui/origin_test.go (hunk 1 above)
Comment (joereuss12) [reply origin_ui/origin_test.go:314]: Added with new commit

**Review by turetske — APPROVED** [review]
LGTM!

**Pull request merged**